packages/core/test/new-api/getParameter.ts

```diff
@@ -0,0 +1,87 @@
+import { assert } from "chai";
+
+import { defineModule } from "../../src/new-api/define-module";
+import { NamedContractDeploymentFutureImplementation } from "../../src/new-api/internal/module";
+import { ModuleConstructor } from "../../src/new-api/internal/module-builder";
+
+describe("getParameter", () => {
+  it("should record given parameters", () => {
+    const constructor = new ModuleConstructor(0, [], { param1: 42 });
+
+    assert.equal(constructor.parameters.param1, 42);
+  });
+
+  it("should allow a parameter to be passed as an arg", () => {
+    const moduleWithParamsDefinition = defineModule("Module1", (m) => {
+      const arg1 = m.getParameter<string>("param1");
```

Comment:
If we do the `= any` that we discussed this `<string>` can be removed, and same in the other non-optional params.